File: backend/auth-service/src/main/java/com/bharatteeka/auth/entity/Patient.java
package com.bharatteeka.auth.entity;

import java.time.LocalDate;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Entity
@Table(name = "patient")
@Data
public class Patient {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "patient_id")
	private Integer patientId;

	@ManyToOne(optional = true, fetch = FetchType.LAZY)
	@JoinColumn(name = "user_id", nullable = true)
	private User user;

	@NotBlank(message = "First name is required")
	@Size(max = 50, message = "First name max 50 characters")
	@Column(name = "first_name", nullable = false, length = 50)
	private String firstName;

	@NotBlank(message = "Last name is required")
	@Size(max = 50, message = "Last name max 50 characters")
	@Column(name = "last_name", nullable = false, length = 50)
	private String lastName;

	@NotNull(message = "Date of birth is required")
	@Column(name = "date_of_birth", nullable = false)
	private LocalDate dateOfBirth;

	@NotBlank(message = "Gender is required")
	@Column(name = "gender", nullable = false)
	private String gender;

	@NotBlank(message = "Aadhaar number is required")
	@Pattern(regexp = "^\\d{12}$", message = "Aadhaar must be exactly 12 digits")
	@Column(name = "aadhar_number", nullable = false, unique = true, length = 12)
	private String aadhaarNumber;

	@NotBlank(message = "Blood group is required")
	@Size(max = 5, message = "Blood group max 5 characters")
	@Column(name = "blood_group", nullable = false, length = 5)
	private String bloodGroup;

	@Column(name = "is_adult")
	private Boolean isAdult = false;

	@Column(name = "is_active")
	private Boolean isActive = true;

	@Column(name = "remarks", columnDefinition = "TEXT")
	private String remarks;
}

File: backend/AdminService/AdminService/Models/Patient.cs
using System;
using System.Collections.Generic;

namespace AdminService.Models;

public partial class Patient
{
    public int PatientId { get; set; }

    public int? UserId { get; set; }

    public string FirstName { get; set; } = null!;

    public string LastName { get; set; } = null!;

    public DateOnly DateOfBirth { get; set; }

    public string Gender { get; set; } = null!;

    public string AadharNumber { get; set; } = null!;

    public string BloodGroup { get; set; } = null!;

    public bool? IsAdult { get; set; }

    public bool? IsActive { get; set; }

    public string? Remarks { get; set; }

    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();

    public virtual ICollection<Notification> Notifications { get; set; } = new List<Notification>();

    public virtual ICollection<ParentChild> ParentChildren { get; set; } = new List<ParentChild>();

    public virtual User? User { get; set; }
}
